phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java

```diff
@@ -3034,6 +3088,11 @@ MutationState dropTable(String schemaName, String tableName, String parentTableN
             return new MutationState(0, 0, connection);
         } finally {
             connection.setAutoCommit(wasAutoCommit);
+            // acquire a mutex on the table to prevent creating views while concurrently
+            // dropping the base table
+            if (tableType == PTableType.TABLE) {
+                deleteCell(null, schemaName, tableName, null);
+            }
```

Comment:
nit: Please update the comment here that you are releasing a mutex.

Reply:
Done.